src/components/menu/menu-item/menu-item.style.js

```diff
@@ -85,11 +87,27 @@ const StyledMenuItemWrapper = styled.a`
       `
     }
 
-    a,
-    ${StyledLink} a,
-    button,
-    ${StyledLink} button {
-      padding: 0 16px;
+    ${
+      !asPassiveItem &&
+      css`
+        a,
+        ${StyledLink} a,
+        button,
+        ${StyledLink} button {
+          padding: 0 16px;
+        }
+      `
+    }
+
+    ${StyledIconButton} {
```

Comment:
I'm not sure about these styles. Are we sure a StyledIconButton will never be used inside a MenuItem in other situations? At the very least I'd check the `asPassiveItem` here as well so this only applies to the situation we're already considering.

Also seeing this made me check the focus styles on the updated code sandbox and while the lack of outline is probably OK, the icon goes dark/invisible when its focused so that will want fixing...

Reply:
Fair point I'll add that as an extra guard, and good spot I've missed the colour override on the Icon 👍